uaclient/tests/test_cli_attach.py

```diff
@@ -202,11 +304,39 @@ def fake_contract_attach(contract_token):
         ret = action_attach(args, cfg)
 
         assert 0 == ret
-        assert 1 == action_status.call_count
+        assert 1 == m_status.call_count
+        assert 1 == m_format_tabular.call_count
         expected_calls = [mock.call(contract_token=token)]
         assert expected_calls == contract_machine_attach.call_args_list
         assert [mock.call(cfg)] == m_update_apt_and_motd_msgs.call_args_list
 
+        # We need to do that since all config objects in this
+        # test will share the same data dir. Since this will
+        # test a successful attach, in the end we write a machine token
+        # file, which will make all other cfg objects here to report
+        # as attached
+        cfg.delete_cache()
```

Comment:
Oof I understand why this is here. Long term, we should try to eliminate scenarios where our unittests write any files to disk - ideally everything would stay in memory and be reset for each test to avoid this kind of problem IMO.

No action necessary for this PR though

Reply:
Yep, I totally agree this is necessary refactor